package/ios/Sources/RNCandle.swift

```diff
@@ -192,211 +279,250 @@ final class HybridRNCandle: HybridRNCandleSpec {
-      return .newYorkTimes
-    case .washingtonPost:
-      return .washingtonPost
-    case .wallStreetJournal:
-      return .wallStreetJournal
-    case .cnn:
-      return .cnn
-    case .yahoo:
-      return .yahoo
-    case .fox:
-      return .fox
-    case .perplexity:
-      return .perplexity
-    case .openai:
-      return .openai
-    case .polymarket:
-      return .polymarket
-    case .espn:
-      return .espn
-    case .youtube:
-      return .youtube
-    case .netflix:
-      return .netflix
+extension TradeQuoteRequest {
+  var toGained: Models.TradeAssetQuoteRequest {
+    get throws {
+      if let fiatAssetQuoteRequest = gained.fiatAssetQuoteRequest {
+        return Models.TradeAssetQuoteRequest.FiatAssetQuoteRequest(
+          .init(
+            assetKind: .fiat,
+            serviceAccountID: fiatAssetQuoteRequest.serviceAccountID,
+            currencyCode: fiatAssetQuoteRequest.currencyCode,
+            amount: fiatAssetQuoteRequest.amount
+          )
+        )
+      } else if let marketAssetQuoteRequest = gained.marketAssetQuoteRequest {
+        return Models.TradeAssetQuoteRequest.MarketAssetQuoteRequest(
+          .init(
+            assetKind: .init(rawValue: marketAssetQuoteRequest.assetKind) ?? .stock,
+            serviceAccountID: marketAssetQuoteRequest.serviceAccountID,
+            serviceAssetID: marketAssetQuoteRequest.serviceAssetID,
+            symbol: marketAssetQuoteRequest.symbol,
+            amount: marketAssetQuoteRequest.amount
+          )
+        )
+      } else if gained.nothingAssetQuoteRequest != nil {
+        return Models.TradeAssetQuoteRequest.NothingAssetQuoteRequest(
+          .init(assetKind: .nothing)
+        )
+      } else if let transportAssetQuoteRequest = gained.transportAssetQuoteRequest {
+        return Models.TradeAssetQuoteRequest.TransportAssetQuoteRequest(
+          .init(
+            assetKind: .transport,
+            serviceAssetID: transportAssetQuoteRequest.serviceAssetID,
+            originCoordinates: transportAssetQuoteRequest.originCoordinates?.toCoordinates,
+            originAddress: transportAssetQuoteRequest.originAddress?.toAddress,
+            destinationCoordinates: transportAssetQuoteRequest.destinationCoordinates?
+              .toCoordinates,
+            destinationAddress: transportAssetQuoteRequest.destinationAddress?.toAddress,
+            seats: transportAssetQuoteRequest.seats
+          )
+        )
+      } else {
+        throw RNClientError.badInitialization(
+          message: "Unsuppoted TradeQuoteRequest: \(String(describing: self))")
+      }
     }
   }
 }
+
+extension Address {
+  var toAddress: Models.Address {
+    return .init(value: value)
+  }
+}
+
+extension Coordinates {
+  var toCoordinates: Models.Coordinates {
+    return .init(
+      latitude: latitude,
+      longitude: longitude
+    )
+  }
+}
+
+extension Models.Service {
+  var toService: Service {
+    Service(fromString: rawValue)!
```

Comment:
#req add back the explicit mapping you had, like I mentioned above for the other enums